File: frameworks/shared/csharp/VelocityBench.HealthCheck/ProbeType.cs
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace VelocityBench.HealthCheck;

/// <summary>
/// Health check probe types (Kubernetes-compatible).
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum ProbeType
{
    [EnumMember(Value = "liveness")]
    Liveness,

    [EnumMember(Value = "readiness")]
    Readiness,

    [EnumMember(Value = "startup")]
    Startup
}

File: frameworks/shared/ruby/lib/velocitybench/healthcheck/probe_type.rb
# frozen_string_literal: true

module VelocityBench
  module HealthCheck
    # Health check probe types (Kubernetes-compatible).
    module ProbeType
      LIVENESS = 'liveness'
      READINESS = 'readiness'
      STARTUP = 'startup'

      ALL = [LIVENESS, READINESS, STARTUP].freeze

      def self.from_string(value)
        normalized = value.to_s.downcase
        return normalized if ALL.include?(normalized)

        raise ArgumentError, "Unknown probe type: #{value}"
      end

      def self.valid?(probe_type)
        ALL.include?(probe_type)
      end
    end
  end
end
